### PR TITLE
fix: paypal setup and key validation on server

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

- Adds possibility to use custom key validations per appId
- Migrates PayPal key validations to server
- Code quality refactoring
- Allows to filter viewer.integrations by appId

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- Install PayPal from App Store
- Input invalid keys, it should fail to create webhook
- Input valid keys, it should succeed and display a success toast

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/app-store/paypal/lib/updateAppCredentials.validator.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  // Test credentials before saving
+  const paypalClient = new Paypal({ clientId: key.client_id, secretKey: key.secret_key });
+  const test = await paypalClient.test();
+  if (!test) throw new Error("Provided credentials failed to authenticate");
+
+  // Delete all existing webhooks
+  const webhooksToDelete = await paypalClient.listWebhooks();
+  if (webhooksToDelete) {
+    const promises = webhooksToDelete.map((webhook) => paypalClient.deleteWebhook(webhook));
```
Comment (alannnc) [inline packages/app-store/paypal/lib/updateAppCredentials.validator.ts:28]: I remember I was going to add some kind of filter to this webhooks so we don't delete any existing webhook from the user.

File: packages/app-store/paypal/lib/updateAppCredentials.validator.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    await Promise.all(promises);
```
Comment (alannnc) [inline packages/app-store/paypal/lib/updateAppCredentials.validator.ts:29]: Should we .catch() here something?

**Review by alannnc — DISMISSED** [review]
Code looks good! Let me just test it.

**Review by zomars — COMMENTED** [review]
Ready for review

**Review by zomars — COMMENTED** [review]
Ready for review

**Review by alannnc — APPROVED** [review]

**Pull request merged**